### ISSUE TITLE
SNS -> SQS Subscription example

### ISSUE THREAD

**Issue by divillysausages** [mj1618/serverless-offline-sns#140]
Is there an example for how to properly set up an SNS -> SQS subscription?

I have a serverless file, like this:

```yaml
plugins:
  - serverless-offline-sns
  - serverless-offline-sqs
  - serverless-offline

provider:
  name: aws
  runtime: nodejs12.x
  region: ${opt:region, 'eu-west-3'}
  stage: ${opt:stage, 'dev'}
  environment:
    ACCOUNT_ID: ${aws:accountId}

custom:
  serverless-offline-sns:
    port: 4002 # a free port for the sns server to run on
    debug: false
    accountId: ${self:provider.environment.ACCOUNT_ID}
    location: .build # Optional if the location of your handler.js is not in ./ (useful for typescript)

functions:
  handleSQSNotifLambda:
    handler: src/handleSQSNotif
    reservedConcurrency: 1 # concurrency control - handle one message at a time
    events:
      - sqs:
          arn:
            Fn::GetAtt:
              - AppSQSQueue
              - Arn
          batchSize: 1 # no race conditions - one message at a time

resources:
  Resources:
    AppSQSQueue:
      Type: AWS::SQS::Queue
      Properties:
        QueueName: ${self:provider.stage}-sqs

    AppSNSTopic:
      Type: AWS::SNS::Topic
      Properties:
        TopicName: ${self:provider.stage}-sns-topic

    AppSNSToSQSSubscription:
      Type: "AWS::SNS::Subscription"
      Properties:
        TopicArn: !Ref AppSNSTopic
        Endpoint: !GetAtt
          - AppSQSQueue
          - Arn
        Protocol: sqs
        RawMessageDelivery: 'true'
```
        
The plugin seems to pick up the `AWS::SNS::Subscription` correctly, though it seems to *link to the final Lambda* as the endpoint, rather than the SQS queue. If I print out the subscriptions, I get

```
[
  {
    SubscriptionArn: 'arn:aws:sns:eu-west-3:000000000000:dev-sns-topic:795517',
    Protocol: 'sqs',
    TopicArn: 'arn:aws:sns:eu-west-3:000000000000:dev-sns-topic',
    Endpoint: 'http://127.0.0.1:4002/dev-handleSQSNotifLambda',
    Owner: '',
    Attributes: { RawMessageDelivery: 'true' },
    Policies: undefined
  }
]
```

If I trace the actual message sent through SQS, it's

```
{
  QueueUrl: 'http://127.0.0.1:4002/dev-handleSQSNotifLambda',
  MessageBody: '{"msg":"hello"}',
  MessageAttributes: {
    msg: { DataType: 'String', StringValue: 'hello' }
  }
}
```

The actual `QueueUrl` should be `http://localhost:9324/000000000000/dev-sqs` (following the `serverless-offline-sqs` format)

The post itself fails - I'm assuming because of the bad endpoint - with the message:

    InvalidChecksum: sendMessage returned an invalid MD5 response. Got "undefined", expecting "b01e24da486a4fe509b187f51145eced".

In https://github.com/mj1618/serverless-offline-sns/issues/134, @bitsofinfo mentions the format

```yaml
  serverless-offline-sns:
    port: 9324
    debug: true
    subscriptions:
      - topic: ${self:provider.environment.INGRESS_EVENT_TOPIC_NAME}
        queue: http://localhost:9324/queue/my-events
```
        
If I add this, then I get *2* subscriptions, one with the correct `QueueUrl` (from the config) and one without (from the `AWS::SNS::Subscription`), though it obviously still crashes because of the wrong one. If I only leave the `subscriptions` list, then I run into the same problem @bitsofinfo had, where the message isn't sent, and there's no way to specify `rawMessageDelivery` in this format.

Fooling around with `remotePort` and `accountId` in the config doesn't help either, as no matter what, the end format still isn't good (it always points to the lambda rather than the queue url)

In short, is there an example somewhere that shows how to correctly set up an SNS -> SQS subscription?

**Comment by bitsofinfo** [comment]
@divillysausages see the PR code in https://github.com/mj1618/serverless-offline-sns/pull/135, if you patch your local w/ that it should work

**Comment by divillysausages** [comment]
Hi @bitsofinfo ,

Just to clarify; your PR fixes the SNS directly sending the message to Lambda, is that correct? As in, it's normal that it bypasses the SQS queue entirely?
